### PR TITLE
Comments: Add method to fetch the latest reply comment ID

### PR DESCRIPTION
**Pull request by dvdchr**
Refs #17087 
Depends on #17399, https://github.com/wordpress-mobile/WordPressKit-iOS/pull/463

As titled, this PR extends Comment Service so it could fetch the latest reply ID from a specified comment ID. In the method parameters, I've added `accountService` to make it testable. As for the `AccountService` usage, the method needs to fetch the current user's dotcom ID to filter returned comments. 

## To test

Since there are no visual changes, make sure all the tests pass.

## Regression Notes
1. Potential unintended areas of impact
n/a. Feature is unreleased.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
n/a. Feature is unreleased.

3. What automated tests I added (or what prevented me from doing so)
Added unit tests for the `CommentService`.

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding unit tests for my changes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WordPress/Classes/Services/CommentService+Replies.swift [hunk 1]
```diff
@@ -0,0 +1,49 @@
+/// Encapsulates actions related to fetching reply comments.
+///
+extension CommentService {
+    /// Fetches the current user's latest reply ID for the specified `commentID`.
```
Comment (ScoutHarris) [inline WordPress/Classes/Services/CommentService+Replies.swift]: Nitpick: This comment doesn't match the parameter name. I assume either this should be `parentID` or the param should be `commentID`

File: WordPress/Classes/Services/CommentService+Replies.swift [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    /// In case if there are no replies found, the success block will still be called with value 0.
```
Comment (ScoutHarris) [inline WordPress/Classes/Services/CommentService+Replies.swift:5]: 👍

**Review by ScoutHarris — APPROVED** [review]
:shipit:

File: WordPress/Classes/Services/CommentService+Replies.swift (hunk 1 above)
Comment (dvdchr) [reply WordPress/Classes/Services/CommentService+Replies.swift]: Oh thank you! I've addressed this in https://github.com/wordpress-mobile/WordPress-iOS/pull/17408/commits/a9379b2cb1386f86853757defcb813898312c100.

Comment (dvdchr) [reply WordPress/Classes/Services/CommentService+Replies.swift]: Oh thank you! I've addressed this in https://github.com/wordpress-mobile/WordPress-iOS/pull/17408/commits/a9379b2cb1386f86853757defcb813898312c100.

**Pull request merged**